OpenRA.Mods.Common/Scripting/Properties/GeneralProperties.cs

```diff
@@ -99,13 +99,15 @@ public class GeneralProperties : ScriptActorProperties
 	{
 		readonly IFacing facing;
 		readonly AutoTarget autotarget;
+		readonly Tooltip tooltip;
 		readonly ScriptTags scriptTags;
 
 		public GeneralProperties(ScriptContext context, Actor self)
 			: base(context, self)
 		{
 			facing = self.TraitOrDefault<IFacing>();
 			autotarget = self.TraitOrDefault<AutoTarget>();
+			tooltip = self.TraitsImplementing<Tooltip>().FirstEnabledTraitOrDefault();
```

Comment:
Don't we want to get the first enabled trait at the time the getter runs?

Reply:
You're right, updated